docs/sdk-and-tools/mxpy/mxpy-cli.md

```diff
@@ -110,17 +112,201 @@ One can alter the current configuration using the command `mxpy config set`. For
 ```sh
-mxpy config set default_address_hrp test
+mxpy env new <name>
+```
+
+Additionally, `--template` can be used to create a config from an existing env config. After a new env config is created, it is set as the active env config. After a new env config is created, the values need to be set using the `mxpy env set` command.
+
+#### Setting the default hrp
+
+The `default_address_hrp` might need to be changed depending on the network you plan on using (e.g Sovereign Chain). Most of the commands that might need the `address hrp` already provide a parameter called `--hrp` or `--address-hrp`, that can be explicitly set, but there are system smart contract addresses that cannot be changed by providing the parameter. If that addresses need to be changed, we can use the following command to set the `default hrp` that will be used throughout mxpy. Here we set the default hrp to `test`:
+
+```sh
+mxpy env set default_address_hrp test
 ```
 
 :::note
 Explicitly providing `--hrp` will **always** be used over the one fetched from the network or the `hrp` set in the config.
 :::
 
+#### Setting the proxy url
+
+If `proxy_url` is set in the active environment, the `--proxy` argument is no longer required for the commands that require this argument.
+
+To set the proxy url, use the following command:
+
+```sh
+mxpy env set proxy_url https://devnet-api.multiversx.com
+```
+
+#### Setting the explorer url
+
+mxpy already knows the explorer urls for all three networks (Mainnet, Devnet, Testnet). This is particularly useful when running the CLI on custom networks where an explorer is also available. This key is not required to be present in the env config for the config to be valid. To set the explorer url use the following command:
+
+```sh
+mxpy env set explorer_url
+```
+
+#### Setting the ask for confirmation flag
+
+If set to `true`, whenever sending a transaction, mxpy will display the transaction and will ask for your confirmation. To set the flag, use the following command:
+
+```sh
+mxpy env set ask_confirmation true
+```
+
+#### Dumping the active env config
+
+We can see the values set in our active env config. To do so, we use the following command:
+
+```sh
+mxpy env dump
+```
+
+#### Dumping all the available env configs
+
+We may have multiple env configs, maybe one for each network. To dump all the available env configs, we use the following command:
+
+```sh
+mxpy env list
+```
+
+#### Deleting a value from the active env config
+
+We can also delete the key-value pairs saved in the env config. For example, let's say we want to delete the explorer url, so we use the following command:
+
+```sh
+mxpy env delete explorer_url
+```
+
+#### Getting a value from the active env config
+
+If we want to see just the value of a specific env config key, we can use the following command:
+
+```sh
+mxpy env get <key-name>
+```
+
+#### Deleting an env config
+
+To delete an env config, we use the following command:
+
+```sh
+mxpy env remove <config-name>
```

Comment:
Oh, now I see that we have both "env delete" and "env remove" :see_no_evil: 

Might be tricky for users...